tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1236,6 +1236,59 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		framework.Logf("nodePort tidbcluster tidb service NodePort haven't changed after update")
 	})
 
+	ginkgo.It("Add heterogeneous cluster into a existing cluster  ", func() {
+		// Create TidbCluster with NodePort to check whether node port would change
+		originTc := fixture.GetTidbCluster(ns, "origin", utilimage.TiDBV3Version)
+		originTc.Spec.PD.Replicas = 1
+		originTc.Spec.TiKV.Replicas = 1
+		originTc.Spec.TiDB.Replicas = 1
+		err := genericCli.Create(context.TODO(), originTc)
+		framework.ExpectNoError(err, "Expected TiDB cluster created")
+		err = oa.WaitForTidbClusterReady(originTc, 30*time.Minute, 15*time.Second)
+		framework.ExpectNoError(err, "Expected TiDB cluster ready")
+
+		heterogeneousTc := fixture.GetTidbCluster(ns, "heterogeneous", utilimage.TiDBV3Version)
+		heterogeneousTc.Spec.PD = nil
+		heterogeneousTc.Spec.TiKV.Replicas = 1
+		heterogeneousTc.Spec.TiDB.Replicas = 1
+		heterogeneousTc.Spec.Cluster = &v1alpha1.TidbClusterRef{
+			Name: originTc.Name,
+		}
+		err = genericCli.Create(context.TODO(), heterogeneousTc)
```

Comment:
I think we should WaitForTidbClusterReady here and then construct the PDClient to check there are two tikv stores, WDYT?